jobs/mysql_backup_job.go

```diff
@@ -123,7 +125,23 @@ func (j *MySQLBackupJob) backup(ctx context.Context) (rerr error) {
 	pipeCtx, cancel := context.WithCancel(ctx)
 	defer cancel()
 
-	xtrabackupCmd := exec.CommandContext(pipeCtx, xtrabackupBin, "--compress", "--backup") // #nosec G204
+	tmpDir, err := ioutil.TempDir("", "mysql-backup")
+	if err != nil {
+		return errors.Wrapf(err, "failed to create tempdir")
+	}
+
+	defer func() {
+		if err := os.RemoveAll(tmpDir); err != nil {
+			j.l.WithError(err).Error("failed to remove temporary directory")
+		}
+	}()
+
+	xtrabackupCmd := exec.CommandContext(pipeCtx,
+		xtrabackupBin,
+		"--compress",
+		"--backup",
+		"--target-dir="+tmpDir,
```

Comment:
It's not required.
It's just another bug. Which is pretty small, so I added it here.

Reply:
So bug in xtrabackup, it creates directory anyway, right? You added this parameter just to track created dir and delete it?

Reply:
Exactly.

I was running pmm-agent as root because of mysql restore...
Then I was rebuilding pmm-agent again using `make install` but it failed, because there was this new folder created by root. And running `make install` as my non-root user failed, because it didn't had access to this dir.

Reply:
Ok, please add a comment describing why we need this dir in the backup job and also report a bug to the `xtrabackup` team.